tests/model_registry/async_job/utils.py

```diff
@@ -1,12 +1,75 @@
 import hashlib
 import requests
 import json
+from kubernetes.dynamic import DynamicClient
+from ocp_resources.job import Job
+from ocp_resources.pod import Pod
+from ocp_resources.service import Service
+from utilities.constants import MinIo
 
+from utilities.general import b64_encoded_string
 from simple_logger.logger import get_logger
 
 LOGGER = get_logger(name=__name__)
 
 
+def check_job_completion(admin_client: DynamicClient, job: Job) -> bool:
+    """
+    Check if job has completed successfully.
+
+    Args:
+        admin_client: Kubernetes dynamic client
+        job: Job object to check
+
+    Returns:
+        bool: True if job completed successfully, False otherwise
+    """
+    job_status = job.instance.status
+
+    if hasattr(job_status, "succeeded") and job_status.succeeded:
+        LOGGER.info("Job completed successfully")
+        return True
+
+    # Log current pod status for debugging
+    try:
+        current_pod = get_latest_job_pod(admin_client=admin_client, job=job)
+        pod_status = current_pod.instance.status
+        if hasattr(pod_status, "phase"):
+            if pod_status.phase == "Succeeded":
+                LOGGER.info(f"Pod {current_pod.name} succeeded")
+            elif pod_status.phase == "Failed":
+                LOGGER.warning(f"Pod {current_pod.name} failed, job may retry with new pod")
+            elif pod_status.phase == "Running":
+                LOGGER.info(f"Pod {current_pod.name} is running...")
+            else:
+                LOGGER.info(f"Pod {current_pod.name} in phase: {pod_status.phase}")
+    except Exception as e:
+        LOGGER.warning(f"Could not get pod status: {e}")
```

Comment:
These are log only. So they would have no impact on your run.

Reply:
yes, thanks for noticing